src/extension/mapml/src/main/java/org/geoserver/mapml/tcrs/WrappingProjType.java

```diff
@@ -0,0 +1,90 @@
+/* (c) 2024 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.mapml.tcrs;
+
+import org.geoserver.mapml.xml.ProjType;
+import org.geotools.api.referencing.FactoryException;
+import org.geotools.api.referencing.crs.CoordinateReferenceSystem;
+import org.geotools.referencing.CRS;
+
+/** Wrap the Projection. In case of a Built-in Projection, it will contain the ProjType enum. */
+public class WrappingProjType {
```

Comment:
I would call it based on what it is, rather than how it's implemented. MapMLProjection maybe?